### PR TITLE
Upgrade Ruma

### LINKED ISSUE #2205: Error parsing error response for `POST /_matrix/client/v3/keys/signatures/upload`
My client is stuck in a loop:

```
rust-crypto: successfully made HTTP request: POST /_matrix/client/v3/keys/signatures/upload
18:24:58.559 react_devtools_backend_compact.js:2367 Error processing outgoing-message requests from rust crypto-sdk Error: deserialization failed: missing field `error` at line 1 column 128
    at push.1685.module.exports.__wbindgen_error_new (matrix_sdk_crypto_js.js:6676:1)
    at <wasm_bindgen::JsError as core::convert::From<E>>::from::h66f6a8cb01617678 (01b16b46:0x313e37)
    at <matrix_sdk_crypto_js::responses::OwnedResponse as core::convert::TryFrom<(matrix_sdk_crypto_js::requests::RequestType,http::response::Response<alloc::vec::Vec<u8>>)>>::try_from::hde971358fe62519b (01b16b46:0x19b93b)
    at olmmachine_markRequestAsSent (01b16b46:0x2118e1)
    at OlmMachine.markRequestAsSent (matrix_sdk_crypto_js.js:3221:1)
    at OutgoingRequestProcessor.makeOutgoingRequest (OutgoingRequestProcessor.ts:99:35)
    at async RustCrypto.outgoingRequestLoop (rust-crypto.ts:876:21)
```

Synapse is returning a response like this:

```json
{
    "failures": {
        "@richvdh:sw1v.org": {
            "EOZDSWJVGZ": {
                "status": 400,
                "errcode": "M_INVALID_SIGNATURE",
                "message": "400: Invalid signature"
            }
        }
    }
}
```

There is no `error` property, and the [spec](https://spec.matrix.org/v1.7/client-server-api/#post_matrixclientv3keyssignaturesupload) does not specify one (though the example rather implies that it should be there).

See also https://github.com/matrix-org/matrix-spec/issues/1594.

### PR DESCRIPTION
**Pull request by jplatte**
Closes #2205.

### CODE REVIEW

**Comment by richvdh** [comment]
Can we get an update in the changelog(s) about what changes we expect this to bring?

**Comment by jplatte** [comment]
It's a tiny incremental update from one git rev to the next, so no, not really.

**Review by poljar — APPROVED** [review]

File: crates/matrix-sdk-crypto/CHANGELOG.md [hunk 1]
```diff
@@ -40,3 +40,6 @@
 - Fix handling of SAS verification start events once we have shown a QR code.
 
 - Fix a bug which could cause generated one-time-keys not to be persisted.
+
+- Fix parsing error for `POST /_matrix/client/v3/keys/signatures/upload`
+  responses generated by Synapse.
```
Comment (richvdh) [inline crates/matrix-sdk-crypto/CHANGELOG.md:45]:
```suggestion
  failure responses generated by Synapse.
```

Comment (jplatte) [reply crates/matrix-sdk-crypto/CHANGELOG.md:45]: It's part of the `failures` key, but it's not actually the endpoint returning with an error so I'd prefer to not change it this way.

**Review by richvdh — APPROVED** [review]
Thank you!

**Pull request merged**